### PR TITLE
Sync stable cadence branch with master

### PR DESCRIPTION
**Pull request by SupunS**
## Description

Conflict resolution: https://gist.github.com/SupunS/cd4fadd7413545dfb5bf44359c504a97
______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
Look good, thank you!

Re: conflict resolution:
- https://gist.github.com/SupunS/cd4fadd7413545dfb5bf44359c504a97#file-sync-stable-cadence-master-2023-10-25-diff-L69-L78: Is this not needed anymore?

File: runtime/interpreter/value.go [hunk 1]
```diff
@@ -16799,12 +16785,53 @@ func (v *CompositeValue) getInterpreter(interpreter *Interpreter) *Interpreter {
...
+	if !ok {
+		return nil
+	}
+
+	var base *EphemeralReferenceValue
+	var self MemberAccessibleValue = v
+	if v.Kind == common.CompositeKindAttachment {
+		base, self = attachmentBaseAndSelfValues(interpreter, v)
+	}
+	return NewBoundFunctionValue(interpreter, function, &self, base, nil)
```
Comment (SupunS) [inline runtime/interpreter/value.go:16834]: IDK for sure, but could it be an issue with passing `nil` as the `boundAuth`?

Comment (turbolent) [reply runtime/interpreter/value.go:16834]: The problem might be that the code at the call-site is now duplicated here: https://github.com/onflow/cadence/pull/2899/files#diff-424575c293dc2bc3a28aab845829d7656fa83a7be4e026de8906935e23bbd52eR16750

Comment (SupunS) [reply runtime/interpreter/value.go:16829]: That seems to be the issue. Updated

**Review by dsainati1 — APPROVED** [review]

**Comment by turbolent** [comment]
@SupunS Haha, yeah! Thanks for the clarification 👍

**Review by turbolent — APPROVED** [review]

**Pull request merged**